### PR TITLE
(FACT-2411) Add debian filesystems fact

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

**Comment by gimmyxd** [comment]
facts tests should be written following: https://github.com/puppetlabs/facter-ng/commit/18cff0a1fa0d5750fdcebfe839df254964000b9a

Also now that spec tests are name spaced based on module/class name there is an inconsistency between the structure in lib and the structure in spec

**Comment by oanatmaria** [comment]
Tests were rewritten according to the example.

File: spec/facter/facts/el/filesystems_spec.rb [hunk 1]
```diff
@@ -2,14 +2,23 @@
...
+    subject(:fact) { Facter::El::Filesystems.new }
+
+    let(:value) { 'ext2,ext3,ext4,xfs' }
+
+    before do
+      allow(Facter::Resolvers::Linux::Filesystems).to \
+        receive(:resolve).with(:systems).and_return(value)
+    end
+
+    it 'calls Facter::Resolvers::Ruby' do
```
Comment (oanatmaria) [inline spec/facter/facts/el/filesystems_spec.rb]: Facter::Resolvers::Linux::Filesystems

File: spec/facter/facts/debian/filesystems_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    subject(:fact) { Facter::Debian::Filesystems.new }
+
+    let(:value) { 'ext2,ext3,ext4,xfs' }
+
+    before do
+      allow(Facter::Resolvers::Linux::Filesystems).to \
+        receive(:resolve).with(:systems).and_return(value)
+    end
+
+    it 'calls Facter::Resolvers::Ruby' do
```
Comment (oanatmaria) [inline spec/facter/facts/debian/filesystems_spec.rb]: Facter::Resolvers::Linux::Filesystems

**Review by Filipovici-Andrei — APPROVED** [review]

**Review by florindragos — APPROVED** [review]

**Pull request merged**